dev/NavigationView/TopNavigationViewDataProvider.cpp

```diff
@@ -69,9 +69,9 @@ int TopNavigationViewDataProvider::IndexOf(const winrt::IInspectable& value)
 {
     if (auto dataSource = m_dataSource.get())
     {
-        auto inspectingDataSource = static_cast<InspectingDataSource*>(winrt::get_self<ItemsSourceView>(dataSource));
+        auto itemsSourceView = winrt::get_self<ItemsSourceView>(dataSource);
```

Comment:
don't need the get_self and the header include above I think.

Reply:
Why did we do that in the firstplace? m_DataSource is already a ItemSourceView? Was that about getting a "different" pointer?

Reply:
Right. we were casting it to the internal type that we know it is so we can get access to the internal method. Now that it is public, we can go through the public api and not have any reference to the implementation class.

---
In reply to: [404155277](https://github.com/microsoft/microsoft-ui-xaml/pull/2231#discussion_r404155277) [](ancestors = 404155277)